docs/DeveloperGuide.md

```diff
@@ -1,22 +1,264 @@
+
+#### Implementation Details
+
+1. **Attributes**:
+    - `String correctAnswer`: Stores the correct answer text for the blank in the question.
+    - `String questionText`: Stores the question text with a placeholder (e.g., `The `____` keyword is used to create a new object in Java.`).
+
+2. **Constructor**:
+    - `FillInTheBlank(String questionText, String correctAnswer)`: Initializes both the question text and correct answer. The constructor ensures that `questionText` includes a blank placeholder and that `correctAnswer` is non-null and non-empty, validating that the question is properly structured.
+
+3. **Methods**:
+    - `checkAnswer(String userAnswer)`:
+      Compares the user’s answer to `correctAnswer`, ignoring case sensitivity to allow for minor capitalization differences. Throws an `IllegalArgumentException` for null or empty inputs, ensuring only valid responses are considered.
+    - `printQuestion()`:
+      Displays the question text with the blank to the user, making it clear where they need to provide an answer.
+    - `toString()`:
+      Returns a formatted string representation of the question, clearly marking it as a fill-in-the-blank question for clarity within the quiz.
+
+#### Design Rationale
+
+- **String Storage for Correct Answer**: The answer is stored as a string to allow precise text matching, which is necessary for fill-in-the-blank questions. This simplifies checking whether the user's input matches the correct answer exactly (or approximately, if extended to support partial matching).
+- **Input Validation in `checkAnswer`**: By ensuring only non-null, non-empty strings are accepted as answers, `checkAnswer` reduces the risk of invalid input and enhances robustness.
+- **Placeholder in Question Text**: Using a placeholder (e.g., `"_____"`) in the question text makes it clear to users where they should mentally "fill in the blank," improving usability.
+
+#### Alternative Considerations
+
+1. **Implementing Case-Sensitive Comparison**: An initial approach was to use case-sensitive comparison for `checkAnswer`, but it was modified to ignore case for a more forgiving user experience.
+2. **Approximate Matching for Answers**: To increase flexibility, an alternative approach could involve approximate matching, allowing answers that closely resemble the correct answer to be accepted (e.g., `"paris"` vs. `"Paris"`). This would, however, add complexity to the validation process and was ultimately deemed unnecessary for this basic implementation.
+
+#### UML Class Diagram
+
+![FillInTheBlank UML Class Diagram](https://github.com/YubotKwng/tp/blob/e3358e1cc869012c202864071fbb75f2f8261712/docs/UML/FITBClassDiagram.png)
```

Comment:
This is a broken link. Please consider adding the image assets to the repo itself to minimize such issues.